### PR TITLE
Add extra product reservation fields

### PR DESCRIPTION
**Pull request by aghimpu**

### CODE REVIEW

File: src/sources/forus-platform/js/angular-1/i18n/i18n-nl.js [hunk 1]
```diff
@@ -1285,9 +1285,40 @@ module.exports = {
...
+            address: "Adres",
+            birth_date: "Geboortedatum",
+            user_note: "Notitie",
         },
     },
 
+    // RESERVATION SETTINGS - reservations-settings.pug
+    reservation_settings: {
+        header: {
+            title: "Reservation settings page",
```
Comment (lexlog) [inline src/sources/forus-platform/js/angular-1/i18n/i18n-nl.js]: needs translation (reservationS settings page)

File: src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            { value: 'xls', label: 'Exporteren XLS', icon: 'file-excel-outline' },
+        ];
+
+        const makeSections = (fields) => ([
+            { type: "radio", key: "data_format", fields: dataFormats, value: "csv", title: 'Kies bestandsformaat' },
+            { type: "checkbox", key: "fields", fields, fieldsPerRow: 3, selectAll: true, title: 'Kies inbegrepen velden' },
+        ]);
+
+        const saveExportedReservations = (data, organization_id, res) => {
+            PushNotificationsService.success('Success!', 'The downloading should start shortly.');
```
Comment (lexlog) [inline src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js]: needs translation and reformulate

File: src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                const { data_format, fields } = data;
+                const queryFilters = { ...filters, data_format, fields };
+
+                PageLoadingBarService.setProgress(0);
+                console.info('- data loaded from the api.');
+
+                ProductReservationService.export(organization_id, queryFilters).then((res) => {
+                    saveExportedReservations(data, organization_id, res);
+                }, (res) => {
+                    PushNotificationsService.danger('Error!', res.data.message);
```
Comment (lexlog) [inline src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js]: needs translation

File: src/sources/forus-platform/pug/tpl/pages/reservation-show.pug [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                    .card-subtitle: .flex
+                        .mdi.mdi-clock-outline
+                        | {{ $ctrl.reservation.created_at_locale }}
+
+                .flex.flex-self-start: .flex-row: .button-group
+                    .button.button-primary.button-sm(
+                        ng-if="$ctrl.reservation.state === 'pending' && !$ctrl.reservation.expired && !$ctrl.reservation.product.deleted" 
+                        ng-click="$ctrl.acceptReservation($ctrl.reservation)")
+                        em.mdi.mdi-check.icon-start
+                        | Accept
```
Comment (lexlog) [inline src/sources/forus-platform/pug/tpl/pages/reservation-show.pug]: needs translation

File: src/sources/forus-platform/pug/tpl/pages/reservation-show.pug [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+                    .button.button-danger.button-sm(
+                        ng-if="$ctrl.reservation.state === 'pending'" 
+                        ng-click="$ctrl.rejectReservation($ctrl.reservation)")
+                        em.mdi.mdi-close.icon-start
+                        | Reject
```
Comment (lexlog) [inline src/sources/forus-platform/pug/tpl/pages/reservation-show.pug]: needs translation

File: src/sources/forus-webshop/pug/tpl/modals/modal-product-reserve-details.pug [hunk 6]
```diff
@@ -4,83 +4,215 @@
...
+
+                    .block-email-setup-details(ng-if="$ctrl.emailShowForm && !$ctrl.emailSubmitted") 
+                        img.block-email-setup-details-logo(src="./assets/img/modal/product-reservation-info.svg")
+
+                        form.form.flex-grow
+                            .form-group.form-group-inline
+                                ui-control-email(ng-model="$ctrl.emailForm.values.email" data-placeholder="bijv. info@info.nl" data-id="email" tabindex="1")
+                                .form-error(ng-repeat="error in $ctrl.emailForm.errors.email" ng-bind="error")
+                        
+                        button.button.button-sm.button-primary(type="submit" ng-click="$ctrl.emailForm.submit()") E-mail submit
```
Comment (lexlog) [inline src/sources/forus-webshop/pug/tpl/modals/modal-product-reserve-details.pug]: needs translation
- [x] @lexlog "E-mail toevoegen"

File: src/sources/forus-webshop/pug/tpl/modals/modal-product-reserve-details.pug [hunk 7]
```diff
@@ -4,83 +4,215 @@
...
+                        .block-email-setup-details-description Momenteel heeft u geen e-mailadres gekoppeld aan uw account.
+
+                        button.button.button-sm.button-primary(ng-click="$ctrl.addEmail()") E-mail toevoegen
+
+                    .block-email-setup-details(ng-if="$ctrl.emailShowForm && !$ctrl.emailSubmitted") 
+                        img.block-email-setup-details-logo(src="./assets/img/modal/product-reservation-info.svg")
+
+                        form.form.flex-grow
+                            .form-group.form-group-inline
+                                ui-control-email(ng-model="$ctrl.emailForm.values.email" data-placeholder="bijv. info@info.nl" data-id="email" tabindex="1")
```
Comment (lexlog) [inline src/sources/forus-webshop/pug/tpl/modals/modal-product-reserve-details.pug]: @lexlog 
- [x] wcag - needs email@email.nl ( everywhere )

**Comment by lexlog** [comment]
hey @Dominique2002, could you take a look on translations above? then you can create commit or write them whatever is easier for you

**Review by lexlog — CHANGES_REQUESTED** [review]
see backend comment

File: src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js (hunk 2 above)
Comment (Dominique2002) [reply src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js]: Success! = Succes!

The downloading should begin shortly. = De download begint over enkele ogenblikken.

(This is based on text already found in the code in VoucherExportService.js)

File: src/sources/forus-platform/js/angular-1/i18n/i18n-nl.js (hunk 1 above)
Comment (Dominique2002) [reply src/sources/forus-platform/js/angular-1/i18n/i18n-nl.js]: Reservation settings page = Reservering instellingen

File: src/sources/forus-platform/pug/tpl/pages/reservation-show.pug (hunk 4 above)
Comment (Dominique2002) [reply src/sources/forus-platform/pug/tpl/pages/reservation-show.pug]: Accept = Accepteer

File: src/sources/forus-platform/pug/tpl/pages/reservation-show.pug (hunk 5 above)
Comment (Dominique2002) [reply src/sources/forus-platform/pug/tpl/pages/reservation-show.pug]: Reject = Weiger

File: src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js (hunk 3 above)
Comment (Dominique2002) [reply src/sources/forus-platform/js/angular-1/services/ProductReservationsExportService.js]: Error = Er is iets fout gegaan.

**Review by lexlog — APPROVED** [review]

**Review by lexlog — APPROVED** [review]

**Pull request merged**